### DISCUSSION TITLE
streaming replication client cannot connect to new primary

### DISCUSSION THREAD

**Discussion by Husselbossel** [cloudnative-pg/cloudnative-pg#3144] [Q&A] [answered]
As @gbartolini pointed out in #2385 

A replication slot exists solely on the instance that created it, and PostgreSQL does not replicate it on the standby servers. As a result, after a failover or a switchover, the new primary does not contain the replication slot from the old primary. Streaming replication clients previously connected to the old primary cannot connect to the new one due to missing replication slots.

Has anyone already solved this issue for themselves and can they point me in a direction to solve it too?
Your help is greatly appreciated!

Jos

**Comment by gbartolini** [comment, marked as answer]
This is still an issue in the backlog. We need to wait for #2835 to be implemented. I am closing this discussion as it should continue in that ticket.

**Reply by Husselbossel** [reply]
got it, thanks!  Any idea on the timeline? 

Ps. thanks for all your work :) great tool
